custom_enumerable_methods.rb

```diff
@@ -0,0 +1,113 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for element in self # rubocop:disable Style/For
+      yield(element)
+    end
+  end
```

Comment:
```rb
2.6.5 :024 > my_each_output = ''
2.6.5 :025 > block = proc { |num, idx| my_each_output += "Num: #{num}, idx: #{idx}\n" }
2.6.5 :026 > array.each_with_index(&block)
 => [3, 4, 2, 6, 8, 4, 6, 5, 1, 2, 4, 1, 5, 6, 7, 8, 1, 7, 3, 0, 8, 0, 3, 6, 0, 1, 0, 6, 2, 6, 3, 7, 3, 2, 3, 5, 1, 1, 7, 8, 2, 3, 4, 8, 1, 8, 0, 3, 4, 5, 4, 3, 3, 7, 1, 8, 4, 7, 8, 2, 6, 8, 0, 6, 0, 5, 1, 2, 5, 2, 2, 4, 0, 2, 1, 8, 1, 3, 3, 8, 2, 8, 1, 6, 1, 0, 1, 2, 5, 0, 5, 8, 1, 3, 2, 7, 4, 5, 2, 4] 
2.6.5 :027 > each_output = my_each_output.dup
2.6.5 :028 > my_each_output = ''
2.6.5 :029 > array.my_each_with_index(&block)
 => [3, 4, 2, 6, 8, 4, 6, 5, 1, 2, 4, 1, 5, 6, 7, 8, 1, 7, 3, 0, 8, 0, 3, 6, 0, 1, 0, 6, 2, 6, 3, 7, 3, 2, 3, 5, 1, 1, 7, 8, 2, 3, 4, 8, 1, 8, 0, 3, 4, 5, 4, 3, 3, 7, 1, 8, 4, 7, 8, 2, 6, 8, 0, 6, 0, 5, 1, 2, 5, 2, 2, 4, 0, 2, 1, 8, 1, 3, 3, 8, 2, 8, 1, 6, 1, 0, 1, 2, 5, 0, 5, 8, 1, 3, 2, 7, 4, 5, 2, 4] 
2.6.5 :030 > my_each_output == each_output
 => false 
2.6.5 :031 >
```

Reply:
Thank you for the test and it seems to be working!

my_each_output = ''
block = proc { |num, idx| my_each_output += "Num: #{num}, idx: #{idx}\n" }
each_output = array.each_with_index(&block)

 my_each_output = ''
 my_each_output = array.my_each_with_index(&block)

p my_each_output == each_output

#true

Reply:
# Thank You! :bouquet: 
## Above issue is fixed now! :heavy_check_mark: 

## I helped us to learn more
### Thank You so much! :bouquet: